frontend/src/components/authentication/SignIn.tsx

```diff
@@ -0,0 +1,124 @@
+/* eslint-disable @typescript-eslint/no-explicit-any */
+/* eslint-disable no-magic-numbers */
+/* eslint-disable react/jsx-no-bind */
+import React, {useState } from 'react'
+import {connect} from 'react-redux'
+import {Dispatch} from 'redux'
+
+import { fetchUser } from '../../actions/userActions'
+import { ActionsOfUser, UserDetailsOnLogin } from '../../actions'
+import {useStyles} from './SignInStyles'
+
+import {CssBaseline, TextField} from '@material-ui/core'
+import {Button, Avatar, Typography, FormControlLabel, Checkbox, Link, Paper, Grid }  from '@material-ui/core'
+import LockOutlinedIcon from '@material-ui/icons/LockOutlined'
+
+export function SignInSide() :React.ReactElement {
+	const initialFormData = Object.freeze({
```

Comment:
- Why do we need `Object.freeze` here?
 - We can move `initialFormData` - move out of react component